src/frontend/components/UploadForm/index.tsx

```diff
@@ -0,0 +1,217 @@
+
+import {
+  notifyUploadProgress,
+  UploadProgressNotification,
+} from '../../data/context/actions';
+import { addResource } from '../../data/genericReducers/resourceById/actions';
+import { RootState } from '../../data/rootReducer';
+import { upload } from '../../data/sideEffects/upload/upload';
+import { appStateSuccess } from '../../types/AppData';
+import { modelName } from '../../types/models';
+import { TimedText, timedTextMode, UploadableObject } from '../../types/tracks';
+import { Maybe } from '../../utils/types';
+import { DASHBOARD_ROUTE } from '../Dashboard/route';
+import { ERROR_COMPONENT_ROUTE } from '../ErrorComponent/route';
+import { IframeHeading } from '../Headings/Headings';
+import { LayoutMainArea } from '../LayoutMainArea/LayoutMainArea';
+import { UploadField } from '../UploadField/UploadField';
+
+const messages = defineMessages({
+  linkToDashboard: {
+    defaultMessage: 'Back to dashboard',
+    description: 'Text for the link to the dashboard in the upload form.',
+    id: 'components.UploadForm.linkToDashboard',
+  },
+});
+
+const titleMessages = defineMessages({
+  [modelName.VIDEOS]: {
+    defaultMessage: 'Create a new video',
+    description: 'Title for the video upload form',
+    id: 'components.UploadForm.title-videos',
+  },
+  [modelName.THUMBNAIL]: {
+    defaultMessage: 'Upload a new thumbnail',
+    description: 'Title for the thumbnail upload form',
+    id: 'components.UploadForm.title-thumbnail',
+  },
+});
+
+const timedtexttrackTitleMessages = defineMessages({
+  [timedTextMode.CLOSED_CAPTIONING]: {
+    defaultMessage: 'Upload a new closed captions file',
+    description: 'Title for the timed text file upload form',
+    id: 'components.UploadForm.title-timedtexttracks-cc',
+  },
+  [timedTextMode.SUBTITLE]: {
+    defaultMessage: 'Upload a new subtitles file',
+    description: 'Title for the timed text file upload form',
+    id: 'components.UploadForm.title-timedtexttracks-st',
+  },
+  [timedTextMode.TRANSCRIPT]: {
+    defaultMessage: 'Upload a new transcript file',
+    description: 'Title for the timed text file upload form',
+    id: 'components.UploadForm.title-timedtexttracks-ts',
+  },
+});
+
+const UploadFormContainer = styled(LayoutMainArea)`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+`;
+
+const IframeHeadingWithLayout = styled(IframeHeading)`
+  flex-grow: 0;
+  margin: 0;
+  text-align: center;
+`;
+
+const UploadFieldContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
+`;
+
+const UploadFormBack = styled.div`
+  line-height: 2rem;
+  padding: 0.5rem 1rem;
+`;
+
+/** Props shape for the BaseUploadForm component. */
+interface BaseUploadFormProps {
+  jwt: string;
+  notifyObjectUploadProgress: (
+    progress: UploadProgressNotification<UploadableObject>['progress'],
+  ) => void;
+  object: Maybe<UploadableObject>;
+  objectType: modelName;
+  updateObject: (object: UploadableObject) => void;
+}
+
+export type Status = Maybe<
```

Comment:
```suggestion
type Status = Maybe<
```

Are there any use sites outside of this file?

Reply:
yes in the `upload` file. You should maybe extract this type somewhere else.

Reply:
I will keep it here.